pkg/controllers/pod_controller_test.go

```diff
@@ -0,0 +1,418 @@
+			jobIdx:            0}).
+			AddAnnotation(batchv1.JobCompletionIndexAnnotation, "0")
+		followerPodWrapper = makePod(&makePodArgs{
+			jobSetName:        jobSetName,
+			replicatedJobName: "replicated-job-1",
+			jobName:           "test-jobset-replicated-job-1-test-job-0",
+			podName:           "test-jobset-replicated-job-1-test-job-0-1",
+			ns:                ns,
+			jobIdx:            0})
+		wantPodWrapper = makePod(&makePodArgs{
+			jobSetName:        jobSetName,
+			replicatedJobName: "replicated-job-1",
+			jobName:           "test-jobset-replicated-job-1-test-job-0",
+			podName:           "test-jobset-replicated-job-1-test-job-0-1",
+			ns:                ns,
+			jobIdx:            0})
+	)
+	tests := []struct {
+		name            string
+		pods            []corev1.Pod
+		wantPodsDeleted []corev1.Pod
+		wantErr         error
+		forceClientErr  bool
+	}{
+		{
+			name: "delete follower pods",
+			pods: []corev1.Pod{
+				leaderPodWrapper.Obj(),
+				followerPodWrapper.Obj(),
+			},
+			wantPodsDeleted: []corev1.Pod{
+				wantPodWrapper.ResourceVersion("999").SetConditions([]corev1.PodCondition{{
+					Type:               corev1.DisruptionTarget,
+					Status:             corev1.ConditionTrue,
+					Reason:             constants.ExclusivePlacementViolationReason,
+					LastTransitionTime: metav1.Now(),
+					Message:            constants.ExclusivePlacementViolationMessage,
+				},
+				}).Obj(),
+			},
+		},
+		{
+			name: "delete follower pods with pod conditions status is false",
+			pods: []corev1.Pod{
+				leaderPodWrapper.Obj(),
+				followerPodWrapper.SetConditions([]corev1.PodCondition{{
+					Type:               corev1.DisruptionTarget,
+					Status:             corev1.ConditionFalse,
+					Reason:             constants.ExclusivePlacementViolationReason,
+					LastTransitionTime: metav1.Now(),
+					Message:            constants.ExclusivePlacementViolationMessage,
+				},
+				}).Obj(),
+			},
+			wantPodsDeleted: []corev1.Pod{
+				wantPodWrapper.ResourceVersion("999").SetConditions([]corev1.PodCondition{{
+					Type:               corev1.DisruptionTarget,
+					Status:             corev1.ConditionTrue,
+					Reason:             constants.ExclusivePlacementViolationReason,
+					LastTransitionTime: metav1.Now(),
+					Message:            constants.ExclusivePlacementViolationMessage,
+				},
+				}).Obj(),
+			},
+		},
+		{
+			name: "delete follower pods with update pod status error",
+			pods: []corev1.Pod{
+				leaderPodWrapper.Obj(),
+				followerPodWrapper.SetConditions([]corev1.PodCondition{{
+					Type:               corev1.DisruptionTarget,
+					Status:             corev1.ConditionFalse,
+					Reason:             constants.ExclusivePlacementViolationReason,
+					LastTransitionTime: metav1.Now(),
+					Message:            constants.ExclusivePlacementViolationMessage,
+				},
+				}).Obj(),
+			},
+			forceClientErr: true,
+			wantErr:        errors.Join(errors.Join(errors.New("example error"))),
```

Comment:
I think there must be a better way to do this than chained joined errors. I'll try to check out this PR locally and figure it out

Reply:
There is a point of view, maybe we should not care about the error type, because it is an error itself, we only need to know `err != nil` in the code. We can not use `assert.Equal` to check whether the type is completely matched, because it is not important

Reply:
Yeah that would be fine

Reply:
I changed the error type to`errors.New("example error")`, and change code to the following
```go
if tc.wantErr != nil {
   assert.Equal(t, tc.wantErr.Error(), gotErr.Error())
}
```
We should focus on the error message rather than whether the types are consistent